code/controllers/subsystem/persistence.dm

```diff
@@ -0,0 +1,109 @@
+#define FILE_RECENT_MAPS "data/RecentMaps.json"
+
+#define KEEP_ROUNDS_MAP 3
+
+//Season defines, universal for all seasons
+#define LAST_UPDATE "last_update" //last time the season changed
+#define CURRENT_SEASON "current_season" //current season of the section
+
+//Season names
+#define SEASONAL_GUNS "seasonal_guns"
+
+SUBSYSTEM_DEF(persistence)
+	name = "Persistence"
+	init_order = INIT_ORDER_PERSISTENCE
+	flags = SS_NO_FIRE
+
+	///Stores how long each season should last
+	var/list/seasons_durations = list(
+		SEASONAL_GUNS = 0.001 DAYS,
+	)
+	///Stores the current season for each season group
+	var/list/season_progress = list()
+	///Items that have been selected for the current round for each season
+	var/list/season_items = list()
+	///Available gun seasons
+	var/static/list/seasons_buckets = list(
+		SEASONAL_GUNS = list(
+		/datum/season_datum/weapons/guns/sadar_event,
+		/datum/season_datum/weapons/guns/wp_event
+		),
+		)
+
+///Loads data at the start of the round
+/datum/controller/subsystem/persistence/Initialize()
+	LoadSeasonalItems()
+	return ..()
+
+///Stores data at the end of the round
+/datum/controller/subsystem/persistence/proc/CollectData()
+	return
```

Comment:
you have plan to use that in the future?

Reply:
Yeah, very likely that something that collects round-end info will be added